### PR TITLE
Shorten GPU names by dropping h. Fix intel Gen 12.7.

### PR DESCRIPTION
**Pull request by garloff**
Plus some typos.

### CODE REVIEW

**Comment by garloff** [comment]
This is for the flavor name generator, where we suggest a short name in addition to a full name.
Strip out the h qualifiers there.

**Comment by mbuechse** [comment]
Shall we only drop the `h` qualifiers or other stuff as well? Dropping `h` alone doesn't do much to shorten the name...

**Comment by garloff** [comment]
Well, I think GPU capacity ("cores") and memory are key properties a user would look for. Also vendor.
We could debate generation, but they are more relevant for GPUs than for CPUs, so I'd rather not drop them...

**Review by mbuechse — COMMENTED** [review]
Oh, I forgot to submit this review.

File: Tests/iaas/flavor-naming/flavor_names.py [hunk 1]
```diff
@@ -250,14 +250,19 @@ def __init__(
 
     def shorten(self):
         """return canonically shortened name as recommended in the standard"""
-        if self.hype is None and self.hwvirt is None and self.cpubrand is None:
+        if self.hype is None and self.hwvirt is None and self.cpubrand is None and self.gpu is None:
             return self
+        # GPU: Remove h modifiers - cleaner would be to build a c'tor like with CPUBrand
```
[the remark this answers is not in the document]
Comment (mbuechse) [reply Tests/iaas/flavor-naming/flavor_names.py]: see e8975f4ab837a72a4487903b1067d8b00ba3762a

**Review by mbuechse — APPROVED** [review]
LGTM (please consider my suggestion in the first review)

File: Tests/iaas/flavor-naming/flavor_names.py [hunk 2]
```diff
@@ -226,13 +244,29 @@ class GPU:
     vram = OptIntAttr("#.V:GiB VRAM")
     vramperf = TblAttr("Bandwidth", {"": "Std BW {<~1GiB/s)", "h": "High BW", "hh": "Very High BW"})
 
+    def __init__(self, gputype="g", brand="N", gen='', cu=None, perf='', vram=None, vramperf=''):
```
Comment (garloff) [inline Tests/iaas/flavor-naming/flavor_names.py:247]: I'd need to double check that we are not defaulting to having a virtualized NVidia GPU this way...

File: Tests/iaas/flavor-naming/flavor_names.py [hunk 3]
```diff
@@ -171,20 +174,29 @@ class Disk:
     disksize = OptIntAttr("#.GB Disk")
     disktype = TblAttr("Disk type", {'': '(unspecified)', "n": "Networked", "h": "Local HDD", "s": "SSD", "p": "HiPerf NVMe"})
 
+    def shorten(self):
+        return self
```
Comment (garloff) [inline Tests/iaas/flavor-naming/flavor_names.py:178]: We could drop disktypes that are networked here, but that would be new functionality... Good to have the option now, but it's a separate discussion, I guess.

File: Tests/iaas/flavor-naming/flavor_names.py (hunk 2 above)
Comment (mbuechse) [reply Tests/iaas/flavor-naming/flavor_names.py:247]: The constructor is only used in one place, namely shorten, and there gputype and brand are propagated.

File: Tests/iaas/flavor-naming/flavor_names.py (hunk 3 above)
Comment (mbuechse) [reply Tests/iaas/flavor-naming/flavor_names.py:178]: Disk type could even be replaced by `''` in any case (even if it's `n`)

**Comment by garloff** [comment]
Thanks, @mbuechse!

**Pull request merged**